scripts/upload_results_to_oep.py

```diff
@@ -0,0 +1,247 @@
+        data_upload_df = pd.read_csv(
+            os.path.join(filepath, filename), encoding="utf8", sep=";", index_col=0
+        )
+
+        metadata = create_metadata(data_upload_df)
+
+        metadata = write_metadata(
+            metadata,
+            SCHEMA,
+            table,
+            f"Model results oemof-B3 {table}",
+            ["RLI", "oemof_b3"],
+        )
+
+        save_dict_to_json(metadata, metadata_path / f"{table}.json")
+
+        logger.info(f"Saved metadata to: {metadata_path}")
+
+    # Creating sql tables from oemetadata
+    # The next command will set up the tables. The collect_tables-function collects all metadata
+    # files in a folder, creates the SQLAlchemy ORM objects and returns them. The tables are
+    # ordered by foreign key. Having a valid metadata strings is necessary for the following steps.
+    tables_orm = oem2orm.collect_tables_from_oem(db, metadata_path)
+
+    # create tables
+    oem2orm.create_tables(db, tables_orm)
+
+    # Writing data into the tables
+    for table, filename in dict_table_filename.items():
+
+        logger.info(f"{filename} is processed")
+
+        data_upload_df = pd.read_csv(
+            os.path.join(filepath, filename), encoding="utf8", sep=";"
+        )
+
+        data_upload_df = data_upload_df.where(pd.notnull(data_upload_df), None)
+
+        # The following command will write the content of your dataframe to the table on the OEP
+        # that was created earlier.
+        # Have a look in the OEP after it ran successfully!
+        logger.info(f"{filename} is written into table {table}")
+
+        try:
+            data_upload_df.to_sql(
+                table,
+                con=db.engine,
+                schema=SCHEMA,
+                if_exists="append",
+                index=False,
+            )
+
+            logger.info("Inserted data to " + SCHEMA + "." + table)
+
+        except Exception as e:
+            logger.error(e)
+            logger.error("Writing to " + table + " failed!")
+            logger.error(
+                "Note that you cannot load the same data into the table twice."
+                " There will be an id conflict."
+            )
+            logger.error(
+                "Delete and recreate with the commands above, if you want to test your"
+                " upload again."
+            )
+
+        logger.info(f"{filename} writing into table ended")
+
+        # Writing metadata to the table
+        # Now that we have data in our table, it is time to metadata to it.
+        md_file_name = f"{table}.json"
+
+        # First we are reading the metadata file into a json dictionary.
+        logger.info(f"{table} read metadata")
+        metadata = oem2orm.mdToDict(
+            oem_folder_path=metadata_path, file_name=md_file_name
+        )
+
+        # Then we need to validate the metadata.
+        logger.info(f"{table} metadata validation")
+        oem2orm.omi_validateMd(metadata)
```

Comment:
i updated this function, you can now activate jsonschema validation by providing the parameter jsonschema_validation=True . This is quite strict and will return a tuple: bool and error dict as well as create a error report in reports/report.json

Reply:
Same here, can do it or move it to an issue it so that we can finish this sooner.